posthog/client.py

```diff
@@ -162,6 +166,165 @@ def sync_execute(query, args=None, settings=None, with_column_types=False):
     return result
 
 
+REDIS_STATUS_TTL = 600  # 10 minutes
+
+
+@dataclass_json
+@dataclass
+class QueryStatus:
```

Comment:
I could see `start_time` and `end_time` being useful here. I'm imagining we want to show query duration to the client or when fetching the results knowing how "fresh" they are.